src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,423 @@
+    restoreTrustStorePathProperty();
+    Properties props = getProperties();
+    addProxyProperties(props);
+    props.put("nonProxyHosts", "*");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyNotUsed();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenNonProxyHostsNotMatchingInProperties() throws SQLException {
+    Properties props = getProperties();
+    addProxyProperties(props);
+    props.put("nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyNotUsedWhenNonProxyHostsMatchingInJVMParams() throws SQLException {
+    restoreTrustStorePathProperty();
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "*");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyNotUsed();
+  }
+
+  @Test
+  public void testProxyUsedWhenNonProxyHostsNotMatchingInJVMParams() throws SQLException {
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  private static void startWiremockStandAlone() {
+    // retrying in case of fail in port bindings
+    await()
+        .alias("wait for wiremock responding")
+        .atMost(Duration.ofSeconds(10))
+        .until(
+            () -> {
+              try {
+                httpProxyPort = findFreePort();
+                httpsProxyPort = findFreePort();
+                wiremockStandalone =
+                    new ProcessBuilder(
+                            "java",
+                            "-jar",
+                            getWiremockStandAlonePath(),
+                            "--root-dir",
+                            System.getProperty("user.dir")
+                                + File.separator
+                                + WIREMOCK_HOME_DIR
+                                + File.separator,
+                            "--enable-browser-proxying", // work as forward proxy
+                            "--proxy-pass-through",
+                            "false", // pass through only matched requests
+                            "--port",
+                            String.valueOf(httpProxyPort),
+                            "--https-port",
+                            String.valueOf(httpsProxyPort),
+                            "--https-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"),
+                            "--ca-keystore",
+                            getResourceURL("wiremock" + File.separator + "ca-cert.jks"))
+                        .start();
+                waitForWiremock();
+                return true;
+              } catch (Exception e) {
+                logger.info(
```

Comment:
it should be error and we can throw the exception since the whole test suit should fail?

Reply:
I changed to warning, we don't want to throw exception but to try again